src/react/reconcilation.js

```diff
@@ -1040,7 +1045,13 @@ export class Reconciler {
   }
 
   _findReactComponentTrees(value: Value, evaluatedNode: ReactEvaluatedNode): void {
-    if (value instanceof AbstractValue) {
+    if (value instanceof BoundFunctionValue) {
```

Comment:
It's not just bound functions. It's all functions that are not ECMAScriptSourceFunctionValue, right?

`value instanceof FunctionValue && !(value instanceof ECMAScriptSourceFunctionValue)`

Reply:
This pattern match should probably be exhaustive. A whitelist instead of blacklist. What values are valid here?